infra/Pos/Communication/Types/Protocol.hs

```diff
@@ -66,6 +70,36 @@ type NSendActions = N.SendActions PackingType PeerData
 newtype ActionSpec m a = ActionSpec (VerInfo -> Action m a)
 type WorkerSpec m = ActionSpec m ()
 
+-- | A NodeId with a type annotation.
+--
+-- TBD should the Ord and Eq instance ignore the peerType?
+-- Probably.
+-- The 'OutboundQueue msg node' should use '(node, NodeType)' for enqueues
+-- but demand an 'Ord' only on 'node'. The current typeclass-based solution
+-- demands that the 'node' itself contains information about its type:
+--
+--   'classifyNode :: node -> NodeType'
+--
+-- and also demands an 'Ord node', but you'd think the 'NodeType' information
+-- wouldn't affect the order.
```

Comment:
@edsko @dcoutts thoughts on this? I find it awkward that I have to include the `NodeType` in the same datatype which will be held in the priority queue.